.github/workflows/unit-tests.yml

```diff
@@ -36,3 +39,27 @@ jobs:
       with:
         name: pytest-coverage-report
         path: coverage.xml
+        
+    - name: Extract coverage percent
+      id: coverage
+      if: github.event_name == 'push'
+      continue-on-error: true
+      run: |
+        fraction=$(sed -n 2p coverage.xml | sed 's/.*line-rate=\"\([0-9.]*\)\".*$/\1/')
+        percentage=$(echo "scale=1; $fraction * 100" | bc -l)
+        percentage_whole=$(echo "${percentage%.*}")
+        colour=$(if [ $percentage_whole -ge 80 ]; then echo "green"; else echo "orange"; fi)
+        echo "::set-output name=colour::$colour"
+        echo "::set-output name=covered::$percentage_whole"
+    
+    - name: Create Coverage Badge
+      uses: schneegans/dynamic-badges-action@v1.2.0
+      if: github.event_name == 'push'
+      continue-on-error: true
+      with:
+        auth: ${{ secrets.GIST_PAT }}
+        gistID: 65d5a42849fd78f4c6e62fad18490d20
```

Comment:
would it be possible to put it to secrets as well?

Reply:
We can, but this is not particularly sensitive information, it just uniquely identifies the Gist, like file name. Also, it's not reusable like PAT. Therefore I'd rather keep it as is, avoiding overcomplicating the secrets.